### DISCUSSION TITLE
Option for exporting migrations including StoredAs parameter on columns

### DISCUSSION THREAD

**Discussion by KirePetrov** [bennett-treptow/laravel-migration-generator#33] [General]
Hi,

Is there any option for exporting migrations including StoredAs parameter on columns?

Now i receive it like normal columns...

Thanks.

**Comment by bennett-treptow** [comment]
Could you give an example of a column with a `storedAs` parameter and it's accompanying MySQL syntax?

e.g. if you have a migration currently that uses one and then also your table's structure using `SHOW CREATE TABLE tablename`?

**Reply by KirePetrov** [reply]
Here is an example of `SHOW CREATE TABLE tablename`

```
CREATE TABLE `tablename` (
`quantity` decimal(24,6) NOT NULL,
`unit_price` decimal(24,6) NOT NULL,
`total` decimal(24,6) GENERATED ALWAYS AS ((`quantity` * `unit_price`)) STORED,
)
```

column `total`  is expored like
`$table->decimal('total', 24, 6);`

missing `->storedAs('quantity * unit_price')`

**Reply by bennett-treptow** [reply]
Check out version 3.2.0 and let me know if that works for you! @KirePetrov

**Reply by KirePetrov** [reply]
> 
> 
> Check out version 3.2.0 and let me know if that works for you! @KirePetrov

@bennett-treptow 
seems like, there is some quote escaping problem.

can you try with this column example?
```
`full_name` varchar(150) COLLATE utf8mb4_unicode_ci GENERATED ALWAYS AS (concat(`first_name`,' ',`last_name`)) STORED,
```

i've got this
![2021-06-09 16_35_15-2021_06_09_163131_create_employees_table php — C__laragon_www_admin — Atom](https://user-images.githubusercontent.com/45404506/121375037-b3689e80-c940-11eb-95a1-ee4a3d72584d.png)

correct one is
`$table->string('full_name', 150)->storedAs("CONCAT(first_name,' ',last_name)");`

**Reply by bennett-treptow** [reply]
Alright give 3.2.1 a try! @KirePetrov

**Reply by KirePetrov** [reply]
> 
> 
> Alright give 3.2.1 a try! @KirePetrov

this causes php error, coz outside and inside quotes should be different. (single/double)

in migration got this
![2021-06-09 17_15_03-2021_06_09_171328_create_employees_table php — C__laragon_www_admin — Atom](https://user-images.githubusercontent.com/45404506/121381908-6ee00180-c946-11eb-8f53-1cae9e13f14f.png)

which can't be processed
![2021-06-09 17_16_53-C__Windows_System32_cmd exe](https://user-images.githubusercontent.com/45404506/121382017-81f2d180-c946-11eb-8f12-7105efa17d6a.png)

**Reply by bennett-treptow** [reply]
Ah didn't even think of that. Alright, 3.2.2 should be the final say here @KirePetrov

**Reply by KirePetrov** [reply]
> 
> 
> Ah didn't even think of that. Alright, 3.2.2 should be the final say here @KirePetrov

thanks, all good now.
